modules/vpc-endpoints/main.tf

```diff
@@ -7,7 +7,7 @@ locals {
 }
 
 data "aws_vpc_endpoint_service" "this" {
-  for_each = local.endpoints
+  for_each = { for k, v in local.endpoints : k => v if lookup(v, "service_name", null) == null }
```

Comment:
I don't think this is necessary to change because the resource `aws_vpc_endpoint.this` will use the `service_name` value if specified anyway.

Reply:
I think we can have it to save a few unnecessary API calls. But maybe it is better to have this data source as extra verification of correctness then I'm going to revert this change.